### PR TITLE
add disposition of component to fe principles

### PR DESCRIPTION
**Pull request by runnez**
trying to find a recommended approach and prevent bloating UI kit in the future

### CODE REVIEW

File: src/frontend.md [hunk 1]
```diff
@@ -221,6 +221,24 @@ export function fakeTicket() {}
 
 1. If some color is repeated several times in code, extract it to theme.
 
+## Disposition of components
+
+- Prefer using local components `(host|mf)/src/components` with a few exceptions
```
Comment (Rendalf) [inline src/frontend.md:226]: `with a few exceptions` what are the exceptions?
maybe you mean `by default`?

File: src/frontend.md [hunk 2]
```diff
@@ -221,6 +221,24 @@ export function fakeTicket() {}
...
+- Use UI Kit distribution if a component represents a design token (1) [websome](<https://www.figma.com/file/I8WglqziPB5pYlFIqF43JE/Client-UI-Kit-(Web)?node-id=0-1&t=MuFhhOjUXP4sFRr2-0>) or matches with [MUI](https://mui.com/material-ui/)
```
Comment (Rendalf) [inline src/frontend.md:227]: what does mean `represents a design token`? do you mean smth like `a component is a part of design guide`?

`or matches with [MUI]` – could you explain how we correlate with MUI?

and why is `agent` ignored here?

File: src/frontend.md [hunk 3]
```diff
@@ -221,6 +221,24 @@ export function fakeTicket() {}
...
+- Use MF distribution if a component is required in several repositories and could be shipped as a whole piece with API
```
Comment (Rendalf) [inline src/frontend.md:228]: `Use MF distribution` – maybe replace with `Expose component from MF`?

`could be shipped as a whole piece with API` – how about to replace with `represents domain of microfrontend and not being common`?

File: src/frontend.md [hunk 4]
```diff
@@ -221,6 +221,24 @@ export function fakeTicket() {}
...
+- Use (websome|agent)-kit distribution if a component is visual and aware domain-specific
+
+```mermaid
+flowchart TD
```
Comment (Rendalf) [inline src/frontend.md:232]: what is meaning of this chart?

File: src/frontend.md [hunk 5]
```diff
@@ -221,6 +221,24 @@ export function fakeTicket() {}
 
 1. If some color is repeated several times in code, extract it to theme.
 
+## Disposition of components
+
+- Prefer using local components `(host|mf)/src/components` with a few exceptions
+- Use UI Kit distribution if a component represents a design token (1) [websome](<https://www.figma.com/file/I8WglqziPB5pYlFIqF43JE/Client-UI-Kit-(Web)?node-id=0-1&t=MuFhhOjUXP4sFRr2-0>) or matches with [MUI](https://mui.com/material-ui/)
+- Use MF distribution if a component is required in several repositories and could be shipped as a whole piece with API
+- Use (websome|agent)-kit distribution if a component is visual and aware domain-specific
```
Comment (Rendalf) [inline src/frontend.md:229]: `if a component is visual` – what do you mean by `visual`?
`aware domain-specific` -> `aware of domain (uses sdk or types from sdk)`?

File: src/frontend.md (hunk 2 above)
Comment (AlekseyLeshko) [inline src/frontend.md:227]: UI-kit. As I remember, we have a de facto rule of triple use. I'd like to add something to the Ui-kit and I'll receive a question how many repos will use it?

File: src/frontend.md (hunk 3 above)
Comment (AlekseyLeshko) [inline src/frontend.md:228]: We have a core functionality and sometimes we want to reuse it. in this case, MF is not a good choice. Websome kit is an example of it.
My experience talks me that we should isolate and group all domains in MFs excluding core, "onboarding" or/and sales.

File: src/frontend.md (hunk 4 above)
Comment (AlekseyLeshko) [inline src/frontend.md:232]: All types of apps can use the UI-kit

Comment (runnez) [reply src/frontend.md]: You are right. This chart is misleading. Deleted it.

File: src/frontend.md (hunk 3 above)
Comment (runnez) [reply src/frontend.md]: Updated

File: src/frontend.md (hunk 2 above)
Comment (runnez) [reply src/frontend.md]: It was before. I deleted it :) 
Let's assume you add a generic component used only in your repo. 
We know exactly what to place in UI Kit if we have the designer's UI kit and MUI as a roadmap,

Comment (runnez) [reply src/frontend.md]: Updated. Take a look, please

File: src/frontend.md [hunk 6]
```diff
@@ -229,8 +229,18 @@ export function fakeTicket() {}
...
+
+1. Place a component into UI Kit if it is added to the designer's shared UI-kit or matches wih MUI components library. [Why MUI?](#ui-kit)
+
+1. Expose a component from MF if it represents a domain of microfrontend and not being common.
+
+1. Place a component into `websome-kit`/`agent-kit` if a component depends on Osome SDK and for some reason it is not desirable to serve it through MF.
+
 ## UI-kit
 
+1. Use `MUI` as the reference for `Component API`. Use **identical** naming for components and props to have a **roadmap** how to scale further.
```
Comment (Rendalf) [inline src/frontend.md:242]: how about adding a reason why we need it?

for example:
`It's needed for having consistent naming and keeping components universal.`

File: src/frontend.md [hunk 7]
```diff
@@ -229,8 +229,18 @@ export function fakeTicket() {}
 
 1. If some color is repeated several times in code, extract it to theme.
 
+## Disposition of components
```
Comment (Rendalf) [inline src/frontend.md:232]: Great rules 👍

**Review by Lecsar — APPROVED** [review]

**Review by Rendalf — APPROVED** [review]

File: src/frontend.md [hunk 8]
```diff
@@ -229,8 +229,18 @@ export function fakeTicket() {}
...
+
+1. Place a component into UI Kit if it is added to the designer's shared UI-kit or matches wih MUI components library. [Why MUI?](#ui-kit)
```
Comment (AlekseyLeshko) [inline src/frontend.md:234]: MUI - Multilingual User Interface?
Could you add a description for this term?

File: src/frontend.md [hunk 9]
```diff
@@ -229,8 +229,18 @@ export function fakeTicket() {}
...
+
+1. Expose a component from MF if it represents a domain of microfrontend and not being common.
+
+1. Place a component into `websome-kit`/`agent-kit` if a component depends on Osome SDK and for some reason it is not desirable to serve it through MF.
+
 ## UI-kit
 
+1. Use `MUI` as the reference for `Component API`. It's needed for having consistent naming and keeping components universal.
```
Comment (divineforest) [inline src/frontend.md:242]: the repo is public, can we rephrase this to be more universal?
